cmd/run.go

```diff
@@ -145,6 +167,50 @@ func getAWSClient(ctx context.Context, conf *config.Config, sess *session.Sessio
 	return awsClient
 }
 
+func getAWSOktaCredentials(profile string) (*credentials.Value, error) {
+
+	config, err := awsokta.NewConfigFromEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	profiles, err := config.Parse()
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := profiles[profile]; !ok {
+		return nil, fmt.Errorf("Profile '%s' not found in your aws config", profile)
+	}
+
+	sessionTTL := time.Hour
+	assumeRoleTTL := time.Hour
+	opts := awsokta.ProviderOptions{
+		// TODO: make MFADevice configurable by config
+		MFADevice:          "phone1",
+		Profiles:           profiles,
+		SessionDuration:    sessionTTL,
+		AssumeRoleDuration: assumeRoleTTL,
+	}
+
+	kr, err := awsokta.OpenKeyring(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err := awsokta.NewProvider(kr, profile, opts)
+	if err != nil {
+		return nil, err
```

Comment:
These can be wrapped with `errors.Wrap(err, "Some more context")` to give a bit more context to users when the errors bubble up